### PR TITLE
feat(DEV-12231): custom Tabs for Receivables 

### PR DESCRIPTION
**Pull request by vkuprin**

### CODE REVIEW

File: packages/sdk-react/mui-styles.d.ts [hunk 1]
```diff
@@ -41,6 +42,7 @@ declare module '@mui/material/styles' {
     MoniteInvoiceRecurrenceIterationStatusChip: 'root';
     MoniteCounterpartStatusChip: 'root';
     MonitePayableTable: 'never';
+    MoniteReceivablesTable: never; // no slots available
```
Comment (vkuprin) [inline packages/sdk-react/mui-styles.d.ts:45]: Please put this in commas quotation marks. Right now it's just text

File: packages/sdk-react/src/components/receivables/ReceivableFilters/ReceivableFilters.tsx [hunk 2]
```diff
@@ -21,69 +18,89 @@ import { DatePicker } from '@mui/x-date-pickers';
...
           >
-            <MenuItem value={undefined}>{t(i18n)`All statuses`}</MenuItem>
+            <MuiInputLabel id="status">{t(i18n)`Status`}</MuiInputLabel>
+            <Select<ReadableReceivablesStatus>
+              labelId="status"
+              label={t(i18n)`Status`}
+              defaultValue={undefined}
+              // value={
+              //   filters.find((filter) => filter.field === 'status')?.value ?? ''
+              // }
```
Comment (vkuprin) [inline packages/sdk-react/src/components/receivables/ReceivableFilters/ReceivableFilters.tsx:83]: Please remove this comments, we have git history

File: packages/sdk-react/src/components/receivables/ReceivableFilters/useReceivablesFilters.tsx [hunk 3]
```diff
@@ -1,23 +1,106 @@
...
+  if (receivableType === 'credit_note') {
+    const creditNoteStatuses = [
+      'draft',
+      'issued',
+    ] satisfies Array<ReceivablesStatusEnum>;
+
+    if (!inStatuses) return creditNoteStatuses;
+
+    return creditNoteStatuses.filter((status) => inStatuses.includes(status));
+  }
```
Comment (vkuprin) [inline packages/sdk-react/src/components/receivables/ReceivableFilters/useReceivablesFilters.tsx:101]: Pleaase refactor to follow little bit more DRY approach

File: packages/sdk-react/src/components/receivables/ReceivablesTable/ReceivablesTable.tsx [hunk 4]
```diff
@@ -74,39 +173,28 @@ const ReceivablesTableBase = ({
...
-
-          <Tab
-            id={`${tabIdPrefix}-${ReceivablesTableTabEnum.CreditNotes}`}
-            aria-controls={`${tabPanelIdPrefix}-${ReceivablesTableTabEnum.CreditNotes}`}
-            label={t(i18n)`Credit notes`}
-            value={ReceivablesTableTabEnum.CreditNotes}
-          />
+          {tabs?.map(({ label }, index) => (
+            <Tab
+              key={index}
```
Comment (vkuprin) [inline packages/sdk-react/src/components/receivables/ReceivablesTable/ReceivablesTable.tsx:183]: I belive the index as a key could be avoided here

File: packages/sdk-react/src/components/receivables/ReceivablesTable/ReceivablesTable.tsx [hunk 5]
```diff
@@ -163,15 +258,37 @@ const ReceivablesTableBase = ({
...
+export const useControlledTab = ({
+  tab: controlledTab,
+  onTabChange: setControlledTab,
 }: Pick<ReceivablesTableProps, 'tab' | 'onTabChange'>) => {
-  const [tabControlled, onTabChangeControlled] =
-    useState<ReceivablesTableTabEnum>(ReceivablesTableTabEnum.Invoices);
+  const [uncontrolledTab, setUncontrolledTab] = useState(controlledTab ?? 0);
+
+  useEffect(() => {
+    if (controlledTab === undefined) return;
```
Comment (vkuprin) [inline packages/sdk-react/src/components/receivables/ReceivablesTable/ReceivablesTable.tsx:276]: Please refactor to `if (!conrolledTab) return`

Comment (PeCrio) [reply packages/sdk-react/src/components/receivables/ReceivablesTable/ReceivablesTable.tsx]: I refactored the logic completely as that doesn't seem to be needed and simplified it to a state and callback.

**Review by PeCrio — APPROVED** [review]

**Pull request merged**